client/packages/system/src/Patient/Encounter/ListView.tsx

```diff
@@ -28,6 +29,12 @@ const EncounterListComponent: FC = () => {
   const { queryParams } = useUrlQueryParams({
     initialSort: { key: 'startDatetime', dir: 'desc' },
   });
+  const patientId = usePatient.utils.id();
+  const filterBy = queryParams.filterBy ?? {};
+  filterBy['patientId'] = {
+    equalTo: patientId,
+  };
+  queryParams.filterBy = filterBy;
```

Comment:
Can you not follow the same pattern as the other ListViews and just include the filter in the main query?:
```ts
const { data, isError, isLoading } = useEncounter.document.list({
    ...queryParams,
    sortBy: {
      key: sortBy.key as EncounterSortFieldInput,
      isDesc: sortBy.isDesc,
      direction: sortBy.isDesc ? 'desc' : 'asc',
    },
    filterBy: { patientId: {equalTo: patientId } }
  });
```

Reply:
Also, probably not a good idea to mutate the `queryParams` object directly, that should only ever be set by the URL (I think)

Reply:
I didn't want to overwrite a potential filter from the query params but just add the patient filter on top...

But yes `queryParams` shouldn't be mutated I, I make a shallow copy now, thanks!

Reply:
> I didn't want to overwrite a potential filter from the query params but just add the patient filter on top...

You could still do
```ts
   filterBy: { ...queryParams.filterBy, patientId: {equalTo: patientId } }
```
within the query?

But what you've got works too.

Reply:
Agree this is simpler will go with your version, thanks!